Arrays/DynamicArray.cpp

```diff
@@ -257,40 +254,196 @@ void DynamicArray::Reverse()
     
 }
 
-void DynamicArray::Swap(int *x, int *y)
+void DynamicArray::Swap(int x, int y)
 {
     int temp;
-    temp = *x;
-    *x = *y;
-    *y = temp;
+    temp = A[x];
```

Comment:
Why are you changing this?

Reply:
Directly passing array elements to a function takes a bit of high time compared to what I've done. If you want to keep the original then I'll do it. Either way is fine

Reply:
Refer to my comment in your code

Reply:
To make the swap function user-friendly I did it so that users can swap elements at any two desired indices. Your algorithm took array elements directly as a parameter which was not ideal. If you want to keep the same old algorithm I'll undo that change. Either way is fine.